src/pal/pal_insert_halt_pass.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef PAL_PLATFORM_INSERT_HALT_PASS_H
+#define PAL_PLATFORM_INSERT_HALT_PASS_H
+
+#include <llvm/IR/PassManager.h>
+#include <llvm/Pass.h>
+
+/// Pass that inserts RDNA specific assembly to halt a shader as soon as it starts if the environment variable
+/// "HALT_IMMEDIATELY" is set to the value "ON"
```

Comment:
What is the purpose of this exactly ? Debugging GPU launch issues ?

Reply:
No, it's used to enable inspecting SGPR values set during work group launch on the GPU right at the beginning of a shader program before any ISA instructions are executed.